src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [userId, setUserId] = useState<number>(0);
+  const [query, setQuery] = useState('');
+  const [filterType, setFilterType] = useState('all');
+
+  const visibleTodos = useMemo(() => {
+    const lowerCaseQuery = query.toLowerCase();
+
+    switch (filterType) {
+      case 'all':
+        return todos.filter(todo => todo.title.toLowerCase().includes(lowerCaseQuery));
+
+      case 'completed':
+        return todos.filter(todo => todo.completed && todo.title.toLowerCase().includes(lowerCaseQuery));
+
+      case 'active':
+        return todos.filter(todo => !todo.completed && todo.title.toLowerCase().includes(lowerCaseQuery));
+
+      default:
+        return todos;
+    }
+  }, [todos, query, filterType]);
+
+  const selectedTodo = todos.find(todo => todo.id === userId);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      try {
+        setTodos(await getTodos());
+      } catch (error) {
```

Comment:
it would be better

```suggestion
      try {
        const todosFromServer = await getTodos();
        
        setTodos(todosFromServer);
      } catch (error) {
```